### PR TITLE
DROOLS-3432 : While using the same template key with difference objects/attributes causes an exception when switching to the Data tab in a Guided Rule Template

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-3432
If the same variable is used for two fields the UI defaults to String cell editor.

PRs:
https://github.com/kiegroup/drools/pull/2332
https://github.com/kiegroup/drools-wb/pull/1134

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-client/src/main/java/org/drools/workbench/screens/guided/template/client/editor/TemplateDataCellFactory.java [hunk 1]
```diff
@@ -63,6 +64,10 @@ public TemplateDataCellFactory(AsyncPackageDataModelOracle oracle,
      */
     public DecoratedGridCellValueAdaptor<? extends Comparable<?>> getCell(TemplateDataColumn column) {
 
+        if(column.getDataType().equals(TemplateModel.DEFAULT_TYPE)){
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-client/src/main/java/org/drools/workbench/screens/guided/template/client/editor/TemplateDataCellFactory.java]: Unfortunately you need to do something similar for the Guided Decision Table's use of "BRL Fragments".

I had a quick look around for _where_ a similar change is needed, but unfortunately my brain let me down.

I'd put a break point in `BRLConditionVariableColumnConverter` and see where it leads.

Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-client/src/main/java/org/drools/workbench/screens/guided/template/client/editor/TemplateDataCellFactory.java]: No worries, good thing you spotted this. I'll look for it.

Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-template-editor/drools-wb-guided-template-editor-client/src/main/java/org/drools/workbench/screens/guided/template/client/editor/TemplateDataCellFactory.java]: BRLConditionColumnPlugin.getDefinedVariables(RuleModel ruleModel) is where I need to add some changes. Adding this here as a reminder for myself ;)

**Comment by jomarko** [comment]
No comment to code, manual check pending.

**Comment by Rikkola** [comment]
@jomarko @manstis These PRs could use a rereview thank you.

**Comment by Rikkola** [comment]
@manstis this needs to go in too :)

**Pull request merged**